app/assets/javascripts/admin/addon/components/theme-settings-editor.js

```diff
@@ -0,0 +1,190 @@
+import Component from "@ember/component";
+import { action, computed } from "@ember/object";
+import { tracked } from "@glimmer/tracking";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+
+export default class ThemeSettingsEditor extends Component {
+  @service dialog;
+
+  @tracked editedContent = null;
+  @tracked errors = [];
+  @tracked saving = false;
+
+  didInsertElement() {
+    this.saving = false;
+    this.errors = [];
+  }
+
+  @computed("editedContent", "saving")
+  get saveButtonDisabled() {
+    return !this.documentChanged || this.saving;
+  }
+
+  @computed("editedContent")
+  get documentChanged() {
+    try {
+      if (!this.editedContent) {
+        return false;
+      }
+      const editedContentString = JSON.stringify(
+        JSON.parse(this.editedContent)
+      );
+      const themeSettingsString = JSON.stringify(this.condensedThemeSettings());
+      const same = editedContentString.localeCompare(themeSettingsString);
+      return same !== 0;
```

Comment:
Nitpick, but could we do

```suggestion
      return editedContentString.localeCompare(themeSettingsString) !== 0;
```